src/Microsoft.VisualStudio.Composition/Configuration/Formatters/ParameterRefFormatter.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.Composition.Formatter
+{
+    using System.Collections.Immutable;
+    using MessagePack;
+    using MessagePack.Formatters;
+    using Microsoft.VisualStudio.Composition.Reflection;
+
+#pragma warning disable CS3001 // Argument type is not CLS-compliant
+
+    public class ParameterRefFormatter : IMessagePackFormatter<ParameterRef?>
```

Comment:
This formatter doesn't appear to need to exist. You can use the `SerializationConstructorAttribute` on the `ParameterRef` constructor that takes the two values you serialize here, and use `[Key(0)]` style attributes on the two fields, and it should work great as a `[MessagePackObject]`. You know, like I see you did for `DiscoveredParts`.